# HG changeset patch
# User David Sterba <dsterba@suse.cz>
# Date 1384521662 -3600
#      Fri Nov 15 14:21:02 2013 +0100
# Branch gentoo-1.7
# Node ID 3d28a71de1f6b90566bddef184bea0e18bc5a8c1
# Parent  fab9a702b51c42fc38fa3e1eefe4944dbd46824d
[PATCH] feature: skip-quoted
From 8c1d030ff5c3e4d9a53d5b0ee5afdc26fbf1adf8 Mon Sep 17 00:00:00 2001
Tunable for number of lines of quoted text that are displayed before the
unquoted text after 'skip to quoted' command (S).

Signed-off-by: David Sterba <dsterba@suse.cz>
---
 globals.h | 1 +
 init.h    | 6 ++++++
 pager.c   | 8 ++++----
 3 files changed, 11 insertions(+), 4 deletions(-)

diff -r fab9a702b51c -r 3d28a71de1f6 PATCHES
--- a/PATCHES	Sat Jan 30 16:13:53 2016 +0000
+++ b/PATCHES	Fri Nov 15 14:21:02 2013 +0100
@@ -1,3 +1,4 @@
+patch-skip-quoted-neomutt
 patch-progress-neomutt
 patch-sidebar-on-right-neomutt
 patch-compress-neomutt
diff -r fab9a702b51c -r 3d28a71de1f6 globals.h
--- a/globals.h	Sat Jan 30 16:13:53 2016 +0000
+++ b/globals.h	Fri Nov 15 14:21:02 2013 +0100
@@ -211,6 +211,7 @@ WHERE short ReflowWrap;
 WHERE short SaveHist;
 WHERE short SendmailWait;
 WHERE short SleepTime INITVAL (1);
+WHERE short SkipQuotedOffset;
 WHERE short TimeInc;
 WHERE short Timeout;
 WHERE short Wrap;
diff -r fab9a702b51c -r 3d28a71de1f6 init.h
--- a/init.h	Sat Jan 30 16:13:53 2016 +0000
+++ b/init.h	Fri Nov 15 14:21:02 2013 +0100
@@ -2875,6 +2875,12 @@ struct option_t MuttVars[] = {
   ** replacing ``%s'' with the supplied string.
   ** For the default value, ``joe'' would be expanded to: ``~f joe | ~s joe''.
   */
+  { "skip_quoted_offset", DT_NUM, R_NONE, UL &SkipQuotedOffset, 0 },
+  /*
+  ** .pp
+  ** Lines of quoted text that are displayed before the unquoted text after
+  ** "skip to quoted" command (S)
+  */
   { "sleep_time",	DT_NUM, R_NONE, UL &SleepTime, 1 },
   /*
   ** .pp
diff -r fab9a702b51c -r 3d28a71de1f6 pager.c
--- a/pager.c	Sat Jan 30 16:13:53 2016 +0000
+++ b/pager.c	Fri Nov 15 14:21:02 2013 +0100
@@ -2307,11 +2307,11 @@ search_next:
 	  int dretval = 0;
 	  int new_topline = topline;
 
-	  while ((new_topline < lastLine ||
+	  while (((new_topline + SkipQuotedOffset) < lastLine ||
 		  (0 == (dretval = display_line (fp, &last_pos, &lineInfo,
 			 new_topline, &lastLine, &maxLine, MUTT_TYPES | (flags & MUTT_PAGER_NOWRAP),
                          &QuoteList, &q_level, &force_redraw, &SearchRE, pager_window))))
-		 && lineInfo[new_topline].type != MT_COLOR_QUOTED)
+		 && lineInfo[new_topline + SkipQuotedOffset].type != MT_COLOR_QUOTED)
 	    new_topline++;
 
 	  if (dretval < 0)
@@ -2320,11 +2320,11 @@ search_next:
 	    break;
 	  }
 
-	  while ((new_topline < lastLine ||
+	  while (((new_topline + SkipQuotedOffset) < lastLine ||
 		  (0 == (dretval = display_line (fp, &last_pos, &lineInfo,
 			 new_topline, &lastLine, &maxLine, MUTT_TYPES | (flags & MUTT_PAGER_NOWRAP),
                          &QuoteList, &q_level, &force_redraw, &SearchRE, pager_window))))
-		 && lineInfo[new_topline].type == MT_COLOR_QUOTED)
+		 && lineInfo[new_topline + SkipQuotedOffset].type == MT_COLOR_QUOTED)
 	    new_topline++;
 
 	  if (dretval < 0)
